src/core/connection.c

```diff
@@ -3116,7 +3116,7 @@ QuicConnRecvHeader(
 
     if (!Packet->IsShortHeader) {
         if (Packet->Invariant->LONG_HDR.Version != Connection->Stats.QuicVersion) {
-            if (Packet->Invariant->LONG_HDR.Version == QUIC_VERSION_VER_NEG) {
+            if (Packet->Invariant->LONG_HDR.Version != QUIC_VERSION_LATEST) {
```

Comment:
This isn't how this is supposed to work. Only `QUIC_VERSION_VER_NEG` is a version negotiation packet. It will contain a list of server supported version numbers. If that list includes `Connection->Stats.QuicVersion` then we should just ignore/drop the packet.